docs/index.md

```diff
@@ -1,3 +1,134 @@
+
+**rocd**
+:   Server component used for stream management and auto discovery over network. Contains a library and a
+    binary that wraps it in a daemon. A stand-alone library is needed for mobile versions with their specific
+    platform limitations (programming language, permission models, ...).
+
+**roc-cast**
+:   Component used for two primary purposes:
+
+    - Provide UI for user interaction with rocd.
+    - On Android (and — possibly — iOS) devices it implements a wrapper application with a so-called main
+      activity context to allow interaction between non-Java libraries (C/C++/...) with mobile sound devices.
+
+**roc-toolkit**
+:   The part that provides some or all of the following components: sound device IO, pipeline, network IO. May
+    be used in binary (MVP) or library forms.
+
+## Usage cases
+
+Here `roc` may be used as a binary only as it doesn't currently implement C API for all 3 components:
+
+![](./assets/dia/simple.svg)
+/// caption
+Simple case: `roc` implements all steps between input and output sound devices
+///
+
+<!-- TODO: add more use case examples here -->
+
+![](./assets/dia/general.svg)
+/// caption
+...
+///
+
+## APIs
+
+`roc-toolkit` may be used in [different ways](https://roc-streaming.org/toolkit/docs/api/reference.html):
+
+![](./assets/dia/roc-toolkit-apis-1.svg)
+/// caption
+as a library for pipeline only (`roc_{sender,receiver}_{encoder,decoder}*`)
+///
+
+![](./assets/dia/roc-toolkit-apis-2.svg)
+/// caption
+as a library for pipeline and network components (`roc_{sender,receiver)*`)
+///
+
+![](./assets/dia/roc-toolkit-apis-3.svg)
+/// caption
+as a library for all three components (`roc_transceiver*`; not implemented)
+///
+
+All three components — `sndio`, `pipeline`, and `netio` — are configured:
+
+| Component | via CLI args | via C API |
+| ---:      | :---:        | :---:     |
+| sndio     | ✅           | ❌        |
+| pipeline  | ✅           | ✅        |
+| netio     | ✅           | ✅        |
+/// caption
+///
+
+## Implementation
+
+`roc-cast` will provide two interfaces:
+- native UI (Flutter) and
+- web UI.
+
+The latter may be used on embedded systems
+
+User configures `roc` (sound devices, network, pipeline) via native `roc-cast` client (primary way) or
+directly by interacting with `rocd` by HTTP API. On mobile platforms it is always `roc-cast` client. On
+embeded platforms the default way is HTTP client.
+
+![](./assets/dia/user--rocd-interaction.svg)
+/// caption
+///
+
+| Platform    | `roc-cast`: native | `roc-cast`: web (HTTP) | `rocd` API (REST) |
+| ---:        | :---:              | :---:                  | :---:             |
+| PC          | ✅ / ✅            | ❌ / ✅                | ❌ / ✅           |
+| Android/iOS | ✅ / ✅            | ❌ / ❌                | ❌ / ✅           |
+| embedded    | ❌ / ❌            | ✅ / ✅                | ❌ / ✅           |
+/// caption
+Default/supported clients for each platform
+///
```

Comment:
I can't understand the meaning of two emojis ("default/supported").

Also, "embedded" targets may support UI, e.g. if we call raspberry pi with a display and keyboard "embedded".

Reply:
cleaned out